io-engine/src/grpc/v1/replica.rs

```diff
@@ -427,4 +429,72 @@ impl ReplicaRpc for ReplicaService {
         )
         .await
     }
+    #[named]
+    async fn create_replica_snapshot(
+        &self,
+        request: Request<CreateReplicaSnapshotRequest>,
+    ) -> GrpcResult<CreateReplicaSnapshotResponse> {
+        self.locked(
+            GrpcClientContext::new(&request, function_name!()),
+            async move {
+                let args = request.into_inner();
+                info!("{:?}", args);
+                let rx = rpc_submit(async move {
+                    let lvol = match UntypedBdev::lookup_by_uuid_str(
+                        &args.replica_uuid,
+                    ) {
+                        Some(bdev) => Lvol::try_from(bdev)?,
+                        None => {
+                            return Err(LvsError::Invalid {
+                                source: Errno::ENOSYS,
+                                msg: format!(
+                                    "Replica {} not found",
+                                    args.replica_uuid
+                                ),
+                            })
+                        }
+                    };
+                    // prepare snap config and flush IO before taking snapshot.
+                    let mut snap_config = match lvol
+                        .prepare_snap_config(&args.snapshot_name)
+                        .await
+                    {
+                        Some(snap_config) => snap_config,
+                        None => {
+                            return Err(LvsError::Invalid {
+                                source: Errno::ENOSYS,
```

Comment:
`ENOSYS` means not implemented, is this the right error code here? 
Please take care considering which error we return and how they map to grpc status so that the control plane may be use them.

Reply:
> `ENOSYS` means not implemented, is this the right error code here? Please take care considering which error we return and how they map to grpc status so that the control plane may be use them.

I have update the error code. Please review once.